examples/mock.rs

```diff
@@ -46,7 +46,9 @@ async fn make_watch_only_events(client: &Client) -> Result<u64> {
 #[tokio::main]
 async fn main() {
     let mock = test::Mock::new();
-    let client = Client::default().with_url(mock.url());
+    // Note that an explicit `with_url` call is not required,
+    // it will be set automatically to the mock server URL.
+    let client = Client::default().with_mock(&mock);
```

Comment:
I'm afraid, such API is very inconvenient to use.

Usually, `url` is passed to some module. Now, the module (and code around) must know about testing env

Reply:
let's address in a follow-up

Reply:
we will continue in #244